### PR TITLE
Implementing Elastic Beanstalk Deployment Script

### PR DESCRIPTION
**Pull request by flomotlik**
Implementing deployment script, installing jq on the system and add an integration test that creates and then deploys to ElasticBeanstalk application/environment

### CODE REVIEW

File: scripts/codeship_aws_eb_deploy [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+APPLICATION_NAME=${2:?'You need to provide the Elastic Beanstalk application name'}
+ENVIRONMENT_NAME=${3:?'You need to provide the Elastic Beanstalk environment name'}
+S3_BUCKET=${4:?'You need to provide the S3 Bucket to upload the artefact to'}
+
+echo "Starting ElasticBeanstalk deployment to $APPLICATION_NAME:$ENVIRONMENT_NAME"
+
+echo "CHANGING Directory to $APPLICATION_FOLDER"
+cd $APPLICATION_FOLDER
+
+ARTEFACT_NAME=${CI_COMMIT_ID}${CI_BUILD_ID}
```
Comment (mlocher) [inline scripts/codeship_aws_eb_deploy]: Are they unique, even if you rerun the deployment? We might want to add the date as well (easier for the user to know which versions to delete, based on the date)

File: scripts/codeship_aws_eb_deploy [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+DEPLOYMENT_FILE_NAME=/tmp/$FILE_NAME
+
+echo "Creating Zip file to deploy in $DEPLOYMENT_FILE_NAME"
+zip -x */.git* -x .git* -x *.hg* -r "$DEPLOYMENT_FILE_NAME" ./
+
+echo "Uploading to S3 for deployment"
+aws s3 cp "$DEPLOYMENT_FILE_NAME" "s3://$S3_BUCKET"
+
+echo "Creating ElasticBeanstalk version: $ARTEFACT_NAME"
+aws elasticbeanstalk create-application-version --application-name "$APPLICATION_NAME" --description "${CI_MESSAGE:0:200}" --version-label "$ARTEFACT_NAME" --source-bundle "S3Bucket=$S3_BUCKET,S3Key=$FILE_NAME"
```
Comment (mlocher) [inline scripts/codeship_aws_eb_deploy]: Limit to the first 200 chars of the first line of the commit message

File: scripts/codeship_aws_eb_deploy (hunk 1 above)
Comment (mlocher) [inline scripts/codeship_aws_eb_deploy]: ARTEFACT_NAME is a misleading name, especially for the later uses as the version

**Comment by mlocher** [comment]
Only checked on my phone, somebody else should take a look as well, but 👍

File: scripts/codeship_aws_eb_deploy (hunk 1 above)
Comment (flomotlik) [inline scripts/codeship_aws_eb_deploy]: Changed to VERSION_NAME and added current time to end of the name

File: scripts/codeship_aws_eb_deploy (hunk 2 above)
Comment (flomotlik) [inline scripts/codeship_aws_eb_deploy]: Isn't that what ${CI_MESSAGE:0:200} is doing? At least locally that works well

**Comment by flomotlik** [comment]
@mlocher could you look through the PR again so we can merge this?

**Comment by mlocher** [comment]
:shipit:

**Pull request merged**